sopel/trigger.py

```diff
@@ -80,6 +82,16 @@ class PreTrigger(object):
         For lines that do *not* contain ``:``, :attr:`text` will be the last
         argument in :attr:`args` instead.
 
+    .. py:attribute:: urls
+        :type: tuple
+
+        The :class:`tuple` of URLs found in the IRC line's text, for
+        ``PRIVMSG`` and ``NOTICE`` only. For other messages, this will be an
+        empty ``tuple``.
+
+        These URLs are found in the text message of the IRC line, usually after
+        the ` :` section.
```

Comment:
This is no longer accurate. It always searches the last parameter now, regardless of whether it's the "text" parameter.

```suggestion
        The :class:`tuple` of URLs found in the IRC line's final parameter.
        This will be an empty ``tuple`` if there are no URLs found.
```

Actually, now that I've made this suggestion… why is it documented both here as a `py:attribute` _and_ below in the `property` docstring? Do we need both, or can we drop down to documenting it in just one place?

Reply:
One of them is for the `PreTrigger` class, the other for the `Trigger` class, which inherit from `str`, not `PreTrigger`.

Reply:
Bah, of course. Then just worry about the wording matching the implementation. 😸

Reply:
Reading the code, it sounds like it matches the implementation: the detection of URLs is done within this `if` block:

```
if self.event == 'PRIVMSG' or self.event == 'NOTICE':
    # [...]
    self.urls = tuple(web.search_urls(self.args[-1], schemes=url_schemes))
```

And in a PRIVMSG/NOTICE, it's usually (not always, then, I guess), after the `:`. I'd like to keep the information about PRIVMSG/NOTICE, what do you suggest?

Reply:
Docs for the `text` attribute already talk about it, so you could just refer there?

Reply:
Oh yes absolutely, good idea!